Realm/Realm/Logging/LogCategory.cs

```diff
@@ -0,0 +1,132 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using Realms.Helpers;
+
+namespace Realms.Logging
+{
+    // TODO(lj): Remove 'abstract' and use this class instead of 'LeafLogCategory'?
```

Comment:
@nirinchev, I could use the `LogCategory` instead of the `LeafLogCategory` but wouldn't it be clearer having the designated leaf class? I don't have a strong opinion on this but the allowed type may indicated more clearly, e.g. compare:
* `public LeafLogCategory Session { get; } = /* can only be a LeafLogCategory here */`
* `public LogCategory Session { get; } = /* can technically be any LogCategory here */`